src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/MarkupCompiler/ParserExtension.cs

```diff
@@ -517,7 +517,7 @@ public override void WriteElementEnd(XamlElementEndNode xamlEndObjectNode)
                         xmlReader.MoveToAttribute(attrName);
                 }
             }
-            else if (!_compiler.IsBamlNeeded && !_compiler.ProcessingRootContext && _compiler.IsCompilingEntryPointClass && xmlReader.Depth > 0)
```

Comment:
This code is called too early to call `_compiler.ProcessingRootContext`.

Basically, there are two steps involved here:
1. Reading a Xaml element.
2. Parsing the Xaml element.

`_compiler.ProcessingRootContext` will only be set in step 2 but this code here is ran in step 1. This means that we get its value before it was set. Since both steps are ran for each Xaml element, when we have 2 or more element we execute this code here with `_compiler.ProcessingRootContext` initialized.

We can remove the check to `_compiler.ProcessingRootContext` because it was already superfluous because the check here `ProcessedRootElement`:
https://github.com/dotnet/wpf/blob/c9a4c6b4a3974df86ffac8d21af1b488c63c630a/src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/MarkupCompiler/ParserExtension.cs#L414
basically does the same check as `_compiler.ProcessingRootContext` but `ProcessedRootElement` is set in step 1 instead of step 2.

I hope my explanation was clear enough to explain my reasoning. Feel free to ask question and I'll try to answer though the Xaml compilation process is very complex and I might not be able to properly explain it.  It makes sense in my head 😄.

Reply:
Thanks for your contribution @ThomasGoulet73 , this change simplifies the code by removing a redundant check to _compiler.ProcessingRootContext, which is called too early in the XAML compilation process to have its value set. The existing check to ProcessedRootElement already serves the same purpose and is set earlier in the process, making the _compiler.ProcessingRootContext check unnecessary. This improves code clarity and avoids potential confusion.
Possibly you can add this in the description as well for better understanding.

Reply:
@anjali-wpf I found it easier to comment on the changes on this specific line. I added a link to this comment in the description to improve clarity.